app/models/manageiq/providers/ibm_cloud/inventory/collector/power_virtual_servers.rb

```diff
@@ -43,6 +43,10 @@ def ports(network_id)
     networks_api.pcloud_networks_ports_getall(cloud_instance_id, network_id).ports || []
   end
 
+  def sap_profiles
+    @sap_profiles ||= sap_api.pcloud_sap_getall(cloud_instance_id)
```

Comment:
Might want to stay consistent with the other collections which cache the resulting array not the whole response so:
`@sap_profiles ||= sap_api.pcloud_sap_getall(cloud_instance_id).profiles || []` 
Then you would do
`collector.sap_profiles.each` in the parser below.